### PR TITLE
Add `dbg()` JS utility function

### PR DESCRIPTION
**Pull request by sbc100**
The idea with this debug-only function is to distinguish actual error messages from debugging tracing.  I think its useful to distinguish between these two cases.

This also allows us to have `err()` always go to console.error and `dbg()` be potentially configurable.  I personally prefer it if log messages don't generate "error"s in the JS console.

This matches llvm's `llvm::dbgs()` stream.

### CODE REVIEW

File: src/runtime_debug.js [hunk 1]
```diff
@@ -106,4 +107,12 @@ function prettyPrint(arg) {
   }
   return arg;
 }
+
+// Used by XXXXX_DEBUG settings to output debug messages.
+function dbg(text) {
+  // Change this to console.error to get backtraces attached to each of the
+  // debug messages.
+  // TODO(sbc): Make this configurable somehow
+  console.warn(text);
```
[the remark this answers is not in the document]
Comment (sbc100) [reply src/runtime_debug.js]: Done

**Comment by Jonathhhan** [comment]
@sbc100 with `ASYNCIFY_DEBUG` I get dbg is undefined. If I add `settings.ASYNCIFY_DEBUG or` here to `emcc.py` `ASYNCIFY_DEBUG` works again:
```
    settings.RUNTIME_DEBUG = (settings.LIBRARY_DEBUG or
                              settings.GL_DEBUG or
                              settings.DYLINK_DEBUG or
                              settings.OPENAL_DEBUG or
                              settings.SYSCALL_DEBUG or
                              settings.WEBSOCKET_DEBUG or
                              settings.SOCKET_DEBUG or
                              settings.ASYNCIFY_DEBUG or
                              settings.FETCH_DEBUG or
                              settings.EXCEPTION_DEBUG or
                              settings.PTHREADS_DEBUG)
```

**Comment by sbc100** [comment]
At yes, that makes sense.  Do you have time to upload a PR by any chance?

**Comment by Jonathhhan** [comment]
@sbc100 done.

**Pull request merged**